### PR TITLE
Replacing PAMELA mesh with VTK mesh

### PR DESCRIPTION
**Pull request by castelletto1**
This PR

- replaces all `msh` meshes by either `vtk` (using ascii format) or `vtu` (using binary format) meshes in [`inputFiles`](https://github.com/GEOSX/GEOSX/tree/develop/inputFiles) folder depending on the model size. Specifically, I kept legacy `vtk` and ASCII for small size problems (less than 1,000 cells) for readability purposes.
- replaces the Egg model `msh` mesh by `vtu` mesh (using binary format) in GEOSXDATA repo

Remaining task/issue:
- [x] updating [tutorial 2](https://geosx-geosx.readthedocs-hosted.com/en/latest/docs/sphinx/tutorials/step02/Tutorial.html) (see issue #2015)
- [x] updating [tutorial 3](https://geosx-geosx.readthedocs-hosted.com/en/latest/docs/sphinx/tutorials/step03/Tutorial.html) in Read the Docs, which will have effects on the [Meshes](https://geosx-geosx.readthedocs-hosted.com/en/latest/coreComponents/mesh/docs/Mesh.html#meshes) documentation in the user Guide (see issue #2015)
- [x] the only parallel integrated test using a mesh that is not internally generated, i.e. `ContactMechanics_SingleFracCompression_03`, produces a `FAIL RUN` because of ghosting problems for fracture elements (see issue #2014)

I don't think we want to keep the PAMELA tests located [here](https://github.com/GEOSX/GEOSX/tree/develop/inputFiles/singlePhaseFlow/pamela_test). @herve-gross please confirm. 

Numerical diffs are introduced due to the different ordering used by PAMELA and VTK mesh generators (visual inspection  confirms that solutions are the same).

GEOSXDATA updating  done in: https://github.com/GEOSX/GEOSXDATA/pull/11
Rebaseline done in https://github.com/GEOSX/integratedTests/pull/244

### CODE REVIEW

**Review by joshua-white — APPROVED** [review]
Looks really great to me.  Nice cleanup.

File: inputFiles/lagrangianContactMechanics/ContactMechanics_PassingCrack_base.xml [hunk 1]
```diff
@@ -138,6 +138,13 @@
       scale="0.0"
       setNames="{ rear }"/>
   </FieldSpecifications>
+ 
+  <Functions>
+    <TableFunction
+      name="apertureTable"
+      coordinates="{ -1.0e-3, 0.0 }"
+      values="{ 0.0, 0.0 }"/>
+  </Functions>
```
Comment (TotoGaz) [inline inputFiles/lagrangianContactMechanics/ContactMechanics_PassingCrack_base.xml:147]: This is not related to your PAMELA/VTK fix, is it?

File: inputFiles/phaseField/PhaseFieldFracture_SpectralSplit.xml [hunk 2]
```diff
@@ -48,21 +48,33 @@
...
+    <Box
+      name="left"
+      xMin="{ -0.01, -0.01, -0.01 }"
+      xMax="{  0.01,  1.01,  0.11 }"/>
+
+    <Box
+      name="right"
+      xMin="{  0.99, -0.01, -0.01 }"
+      xMax="{  1.01, 1.01,  0.11 }"/>
+  </Geometry>
```
Comment (TotoGaz) [inline inputFiles/phaseField/PhaseFieldFracture_SpectralSplit.xml:76]: Why do you have to add this?

File: inputFiles/phaseField/PhaseFieldFracture_VolDevSplit.xml [hunk 3]
```diff
@@ -48,11 +48,33 @@
...
+    <Box
+      name="left"
+      xMin="{ -0.01, -0.01, -0.01 }"
+      xMax="{  0.01,  1.01,  0.11 }"/>
+
+    <Box
+      name="right"
+      xMin="{  0.99, -0.01, -0.01 }"
+      xMax="{  1.01, 1.01,  0.11 }"/>
+  </Geometry>
```
Comment (TotoGaz) [inline inputFiles/phaseField/PhaseFieldFracture_VolDevSplit.xml:76]: Same question

**Review by TotoGaz — APPROVED** [review]
So what about the PAMELA code?
When are we going to stop linking to it?

> the only parallel integrated test using a mesh that is not internally generated, i.e. ContactMechanics_SingleFracCompression_03, produces a FAIL RUN because of ghosting problems for fracture elements.

The tests are supposed to protect the features of `GEOSX`.
Unfortunately, this feature in parallel is notoriously buggy and very hard to fix 🤷 😢 (it would require a dedicated person at least and I'm afraid we do not have the resources right now).
If you cannot make it work at minimal cost, I'd say we can suspend this test as it's not the protector of any reliable feature.

> I don't think we want to keep the PAMELA tests located [here](https://github.com/GEOSX/GEOSX/tree/develop/inputFiles/singlePhaseFlow/pamela_test).

IMHO eventually, this should disappear too.

File: inputFiles/lagrangianContactMechanics/ContactMechanics_PassingCrack_base.xml (hunk 1 above)
Comment (castelletto1) [reply inputFiles/lagrangianContactMechanics/ContactMechanics_PassingCrack_base.xml:147]: Yes, this is unrelated. `ContactMechanics_PassingCrack` was removed from the integrated tests. As a result, the `xml` file was not updated to ensure that attribute `apertureTableName` was not empty. I decided to modify to make sure it can run.

As a side comment, specifying an aperture makes sense when contact mechanics is coupled with fluid flow. For pure contact mechanics problem this field is irrelevant and should not be a required input.

File: inputFiles/phaseField/PhaseFieldFracture_SpectralSplit.xml (hunk 2 above)
Comment (castelletto1) [reply inputFiles/phaseField/PhaseFieldFracture_SpectralSplit.xml:76]: `top`, `bottom`, `left` and `right` were attributes associated to faces in the original `msh` file. In the `vtk ` mesh I got rid of cell attributes and simply kept three-dimensional cells with the above boxes introduced to identify boundary nodes.

File: inputFiles/phaseField/PhaseFieldFracture_VolDevSplit.xml (hunk 3 above)
Comment (castelletto1) [reply inputFiles/phaseField/PhaseFieldFracture_VolDevSplit.xml:76]: See answer above.

**Comment by castelletto1** [comment]
> The tests are supposed to protect the features of `GEOSX`.
> Unfortunately, this feature in parallel is notoriously buggy and very hard to fix 🤷 😢 (it would require a dedicated person at least and I'm afraid we do not have the resources right now).
> If you cannot make it work at minimal cost, I'd say we can suspend this test as it's not the protector of any reliable feature.

OK, I propose we disable the parallel test and open an issue to remember to reactive it, most likely when issue #841 will be resolved.

**Comment by TotoGaz** [comment]
> OK, I propose we disable the parallel test and open an issue to remember to reactive it, most likely when issue #841 will be resolved.

I'm fine with this. Still require additional mgmt validation though.

File: src/docs/sphinx/advancedExamples/validationStudies/faultMechanics/singleFracCompression/Example.rst [hunk 4]
```diff
@@ -76,7 +76,7 @@ The following figure shows the mesh used in this problem.
 
 Here, we load the mesh with ``PAMELAMesh`` (see :ref:`ImportingExternalMesh`).
 The syntax to import external meshes is simple: in the XML file,
```
Comment (jhuang2601) [inline src/docs/sphinx/advancedExamples/validationStudies/faultMechanics/singleFracCompression/Example.rst:78]:
```suggestion
Here, we load the mesh with ``VTKMesh``.
The syntax to import external meshes is simple: in the XML file,
```

**Review by jhuang2601 — APPROVED** [review]
